### ISSUE TITLE
Git - Support SSH keys with a hardware key

### ISSUE THREAD

**Issue by lszomoru** [microsoft/vscode#161782]
In September 2022 we have added support for SSH keys with a passphrase (https://github.com/microsoft/vscode/issues/13680).
This issue is tracking the work to add support for SSH keys that are associated with a hardware key.

**Comment by neo** [comment]
Is this related to #159584?

**Comment by bcvanmeurs** [comment]
Is this related to this error message, when using the GitHub Pull Requests and Issues Extension?
> An error occurred when fetching the repository: sign_and_send_pubkey: signing failed for ECDSA-SK "~/.ssh/id_ecdsa_sk": device not found git@github.com: Permission denied (publickey). fatal: Could not read from remote repository. Please make sure you have the correct access rights and the repository exists.

**Comment by glitchwizard** [comment]
This is closed, and I got here from this issue; https://github.com/microsoft/vscode/issues/13680

**Comment by glitchwizard** [comment]
@lszomoru  so has #13680 been pushed to the normal VScode release? Not just the insider version? That thread is locked to collaborators only so here's the only place I found where I can look into this.

**Comment by lszomoru** [comment]
@glitchwizard, yes, the capability to interact with a git remote using an SSH key with a passphrase has been released to Stable for several milestones now. Having said that there are several scenarios that are not supported yet:
* Interact with a git remote using an SSH key with a passphrase stored on a hardware key
* Signing a commit with an SSH key with a passphrase

**Comment by glitchwizard** [comment]
@lszomoru The issue I'm running into though is that if you use WSL, it doesn't ask for the passphrase to the key, and then hangs the commit. I can only actually push via the terminal, not in VS Code itself.

**Comment by umagon** [comment]
@lszomoru I've been having this problem for ages now.

**Comment by jorritvanderheide** [comment]
+1
